chart/kubeapps/templates/kubeapps-frontend-config.yaml

```diff
@@ -57,36 +57,40 @@ data:
       # Ensure each cluster can be reached (should only be
       # used with an auth-proxy where k8s credentials never leave
       # the cluster). See clusters option.
-      {{- range .Values.clusters }}
+    {{- range .Values.clusters }}
       location ~* /api/clusters/{{ .name }} {
-        rewrite /api/clusters/{{ .name }}/(.*) /$1 break;
-        rewrite /api/clusters/{{ .name }} / break;
-
-      {{/* Helm returns a nil pointer error when accessing foo.bar if foo doesn't
+    {{/* We need to split the API service URL(s) into the base url and the path segment so
+           those configurations using a path can be appropriately rewritten below while
+           ensuring the proxy_pass statement is given the base URL only. */}}
+    {{- $apiServiceURLMap := include "kubeapps.parseAPIServiceURL" .apiServiceURL | fromYaml }}
```

Comment:
@absoludity and here is the `| fromYaml` usage, whereby I consume the YAML string output of my named template `kubeapps.parseAPIServiceURL` and cast it to a dict which is then assigned for usage/reference in this file, e.g. `$apiServiceURLMap.path`.

Reply:
Yeah, given it's just two values, seems a bit odd to go through the yaml necessities (I forgot the helpers are evaluated as strings only).

We could just do two separate helpers, but at that point, given they're not re-used, it may just be simpler to do it inline as you'd done originally (sorry). To save you the hassle, I've redone it here, and also added sane defaults so it works when an `.apiServiceURL` isn't defined: https://github.com/mecampbellsoup/kubeapps/pull/1